src/Build/Instance/TaskRegistry.cs

```diff
@@ -660,7 +671,14 @@ private void RegisterTask(string taskName, AssemblyLoadInfo assemblyLoadInfo, st
                 _taskRegistrations[taskIdentity] = registeredTaskEntries;
             }
 
-            registeredTaskEntries.Add(new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord));
+            RegisteredTaskRecord newRecord = new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord);
+
+            if (overrideTask && !overriddenTasks.ContainsKey(taskName))
```

Comment:
It seems like you should be able to override an override.
```suggestion
            if (overrideTask)
```
Maybe a warning?

Reply:
When would this happen? Some complex `Condition` evaluation that a user thought was exhaustive but didn't quite catch one particular corner case?  It's my understanding that most `<UsingTask>`s are done in a block, with conditions and such to make sure that the best one is chosen. That would mean multiple `Overrides` applying would be unexpected.